server/models/movement.js

```diff
@@ -15,4 +20,34 @@ const movementSchema = new Schema({
   timestamp: { type: mongoose.Schema.Types.Date, default: Date.now },
 });
 
+//Post saving check the car location and the movement location
+movementSchema.post("save", async function(doc, next) {
```

Comment:
can we move this application logic out of the schema? it could either be in a helper method or directly in the controller method, but it seems way beyond the scope of a model to be checking against other models and sending emails